templates/batch-llm/README.md

```diff
@@ -1,326 +1,199 @@
-            generated_text.append(' '.join([o.text for o in output.outputs]))
-        return {
-            "prompt": prompt,
-            "generated_text": generated_text,
-        }
 ```
 
-### Scaling with GPUs
 
-Next, apply batch inference for all input data with the Ray Data [`map_batches`](https://docs.ray.io/en/latest/data/api/doc/ray.data.Dataset.map_batches.html) method. When using vLLM, LLM instances require GPUs; here, we will demonstrate how to configure Ray Data to scale the number of LLM instances and GPUs needed.
 
-To use GPUs for inference in the Workspace, we can specify `num_gpus` and `concurrency` in the `ds.map_batches()` call below to indicate the number of LLM instances and the number of GPUs per LLM instance, respectively. For example, if we want to use 4 LLM instances, with each requiring 1 GPU, we would set `concurrency=4` and `num_gpus=1`, requiring 4 total GPUs.
+## Step 3: Run the batch inference through RayLLMBatch
 
-Smaller models, such as `Meta-Llama-3-8B-Instruct` and `Mistral-7B-Instruct-v0.1`, typically require 1 GPU per instance. Larger models, such as `Mixtral-8x7B-Instruct-v0.1` and `meta-llama/Meta-Llama-3-70B-Instruct`, typically require multiple GPUs per instance. You should configure these parameters according to the compute needed by the model.
 
+With the workload and model configs defined, we can now run the batch inference through RayLLMBatch.
 
 ```python
-ds = ds.map_batches(
-    LLMPredictor,
-    # Set the concurrency to the number of LLM instances.
-    concurrency=4,
-    # Specify the number of GPUs required per LLM instance.
-    num_gpus=1,
-    # Specify the batch size for inference. Set the batch size to as large as possible without running out of memory.
-    # If you encounter out-of-memory errors, decreasing batch_size may help.
-    batch_size=5,
-    # Pass keyword arguments for the LLMPredictor class.
-    fn_constructor_kwargs={"text_column": "item"},
-    # Select the accelerator type; A10G or L4.
-    accelerator_type=get_a10g_or_equivalent_accelerator_type(),
-)
-```
 
-Finally, make sure to either enable *Auto-select worker nodes* or configure your workspace cluster to have the appropriate GPU worker nodes (A10G or L4):
+from rayllm_batch import RayLLMBatch
 
-<img src="https://raw.githubusercontent.com/anyscale/templates/main/templates/batch-llm/assets/ray-data-gpu.png"/>
+batch = RayLLMBatch(
+    engine_config=engine_config,
+    workload=workload,
+    # Specify the batch size for inference. Set the batch size to as large as possible without running out of memory.
+    # If you encounter out-of-memory errors, decreasing batch_size may help. 
+    batch_size=None,
+    # Set the number of replicas to use for the inference. Each replica will run one instance of inference pipeline.
+    num_replicas=1,
+)
 
-Run the following cell to start dataset execution and view the results!
 
+# This will runs until completion.
+ds = batch.run()
```

Comment:
```suggestion
# This will runs until completion. If you specify output_path=..., then
# the results will be written to local disk (local://) or AWS S3 (s3://).
# In this example, we only keep results in memory for demo purpose.
ds = batch.run()
```